src/states/node.rs

```diff
@@ -479,24 +490,15 @@ impl Node {
                 return;
             }
         } else {
```

Comment:
There are two `if`s inside the single `else` block though... maybe I'm missing something?